sqldelight-sample/src/main/java/com/example/sqldelight/hockey/data/HockeyOpenHelper.kt

```diff
@@ -0,0 +1,97 @@
+package com.example.sqldelight.hockey.data
+
+import android.arch.persistence.db.SupportSQLiteDatabase
+import android.arch.persistence.db.SupportSQLiteOpenHelper
+import android.content.Context
+import com.example.sqldelight.hockey.QueryWrapper
+import com.example.sqldelight.hockey.data.PlayerVals.Position
+import com.example.sqldelight.hockey.data.PlayerVals.Shoots
+import com.squareup.sqldelight.EnumColumnAdapter
+import com.squareup.sqldelight.android.create
+import com.squareup.sqldelight.db.SqlDatabase
+import java.util.GregorianCalendar
+
+object HockeyOpenHelper : SupportSQLiteOpenHelper.Callback(QueryWrapper.version) {
+  private var instance: QueryWrapper? = null
+
+  private fun createQueryWrapper(database: SqlDatabase): QueryWrapper {
+    return QueryWrapper(
+        database = database,
+        teamAdapter = Team.Adapter(DateAdapter()),
+        playerAdapter = Player.Adapter(
+            birth_dateAdapter = DateAdapter(),
+            shootsAdapter = EnumColumnAdapter(),
+            positionAdapter = EnumColumnAdapter()
+        )
+    )
+  }
+
+  fun getInstance(context: Context): QueryWrapper {
+    if (instance == null) {
+      instance = createQueryWrapper(QueryWrapper.create(
```

Comment:
yea absolutely. Probably the clunkiest part of the api right now. It's tricky because the companion object for `QueryWrapper` is also the db lifecycle callback handler (it has `onMigrate` and `onCreate`) and so it needs to be used when creating the database. Thats what `QueryWrapper.create()` does. But its a dumb looking api because you would think it creates a `QueryWrapper`, when really it creates a `SqlDatabase` which then can be used to make a `QueryWrapper`

I'm very open to suggestions on how to make this nicer. I'll put more thought on it next week.